scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py

```diff
@@ -149,31 +152,35 @@ def notify_data_changed(self):
     # ------------------------------------------------------------------------------------------------------------------
     # Adding / removing table entries
     # ------------------------------------------------------------------------------------------------------------------
-
     def add_entry_to_table(self, row_entries):
         assert len(row_entries) == self.grouping_table.columnCount()
-
         row_position = self.grouping_table.rowCount()
         self.grouping_table.insertRow(row_position)
         for i, entry in enumerate(row_entries):
             item = QtWidgets.QTableWidgetItem(entry)
-            if group_table_columns[i] == group_table_columns[0]:
+            if group_table_columns[i] == 'group_name':
                 # column 0 : group name
                 group_name_widget = table_utils.ValidatedTableItem(self._validate_group_name_entry)
                 group_name_widget.setText(entry)
                 self.grouping_table.setItem(row_position, 0, group_name_widget)
                 self.grouping_table.item(row_position, 0).setToolTip(entry)
-                item.setFlags(QtCore.Qt.ItemIsEnabled)
-                item.setFlags(QtCore.Qt.ItemIsSelectable)
-            if group_table_columns[i] == group_table_columns[1]:
-                # column 1 : detector IDs
+            if group_table_columns[i] == 'to_analyse':
+                # columns 1 : whether to analyse this group
+                if entry:
+                    item.setCheckState(QtCore.Qt.Checked)
+                else:
+                    item.setCheckState(QtCore.Qt.Unchecked)
+            if group_table_columns[i] == 'detector_ids':
+                # column 2 : detector IDs
                 detector_widget = table_utils.ValidatedTableItem()
                 detector_widget.setText(entry)
-                self.grouping_table.setItem(row_position, 1, detector_widget)
-                self.grouping_table.item(row_position, 1).setToolTip(entry)
+                self.grouping_table.setItem(row_position, 2, detector_widget)
```

Comment:
Could we define these column numbers as variables at the top of the file to reduce chances of error later and make the code clearer.